_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -2014,7 +2035,14 @@ mfxStatus ImplementationAvc::SubmitToMctf(DdiTask * pTask, bool isSceneChange, b
         if (pData.Y == 0)
             return Error(MFX_ERR_LOCK_MEMORY);
 
-        MFX_SAFE_CALL(amtMctf->MCTF_PUT_FRAME(pData.Y, isSceneChange, false, nullptr, doIntraFiltering));
+        MFX_SAFE_CALL(amtMctf->MCTF_PUT_FRAME(
+            pData.Y,
+            pTask->m_handleMCTF,
+            (void**)&pTask->m_cmMCTF,
```

Comment:
what's the reason to have type conversion here? Can we just declare MCTF_PUT_FRAME in a way proper type will be exposed?

Reply:
Yes, I am cleaning this up